### PR TITLE
[explorer] show source code for verified modules

### PR DESCRIPTION
**Pull request by williamrobertson13**
## Description 
This PR implements phase 1 of our source code verification functionality according to https://www.notion.so/mystenlabs/Source-Code-Verification-1aaebb21554e466eaf70e19db2f66825.  To summarize, we're going to show a "Source [Verified]" tab for verified modules as determined by our new `source.mystenlabs.com/api` endpoint. I'll get sign off from @rvantonder and Sean before landing :)

Verified source tab:
https://explorer-git-wrobertson-sourcever-mysten-labs.vercel.app/object/0x0000000000000000000000000000000000000000000000000000000000000002

No source tab:
https://explorer-git-wrobertson-sourcever-mysten-labs.vercel.app/object/0x04e20ddf36af412a4096f9014f4a565af9e812db9a05cc40254846cf6ed0ad91

## Test Plan 
- Manual testing on devnet/testnet/mainnet against verified and non-verified modules
- Manual testing on mobile/desktop

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: apps/explorer/src/components/module/PkgModulesWrapper.tsx [hunk 1]
```diff
@@ -7,66 +7,47 @@ import clsx from 'clsx';
...
 	const [searchParams, setSearchParams] = useSearchParamsMerged();
 	const [query, setQuery] = useState('');
 
-	// Extract module in URL or default to first module in list
-	const selectedModule =
-		searchParams.get('module') && modulenames.includes(searchParams.get('module')!)
-			? searchParams.get('module')!
-			: modulenames[0];
+	const moduleNameValue = searchParams.get('module');
+	const moduleFromParams = moduleNameValue
```
Comment (williamrobertson13) [inline apps/explorer/src/components/module/PkgModulesWrapper.tsx:33]: I refactored some of this for better type safety / easier readability / easier data plumbing

**Review by pchrysochoidis — APPROVED** [review]

**Review by rvantonder — APPROVED** [review]
Awesome, LGTM! I have some questions about the feature flagging that I'll ask on slack :)

File: apps/explorer/src/hooks/useVerifiedSourceCode.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+		queryFn: async () => {
+			const response = await fetch(
+				`https://source.mystenlabs.com/api?network=${network.toLowerCase()}&address=${packageId}&module=${moduleName}`,
+			);
+			if (!response.ok) {
+				throw new Error(`Encountered unexpected response: ${response.status}`);
+			}
+
+			const jsonResponse: UseVerifiedSourceCodeResponse = await response.json();
+			return jsonResponse.source || null;
```
Comment (rvantonder) [inline apps/explorer/src/hooks/useVerifiedSourceCode.ts:43]: Belated comment to clarify how we handle errors: Right now if the server returns an error

- We don't show the source tab whenever there's an error in the response
- We don't do anything with the error right now (I think that's OK, the important bit right now is to not display source tabs if there is an error)

Comment (williamrobertson13) [reply apps/explorer/src/hooks/useVerifiedSourceCode.ts:43]: Exactly! We'll only show the source tab when the API returns a successful response with a `source` property 👍🏼

**Pull request merged**